Move plone locking info to plone app layout

cihanandac opened the pull request:
Note by @ale-rt:
<mark>Before merging we need a PR that removes the moved code from plone.app.layout</mark>

Requires:

- [ ] https://github.com/zopefoundation/z3c.jbot/pull/20
- [ ] https://github.com/plone/plone.locking/pull/45

Refs:

- [ ] https://github.com/plone/Products.CMFPlone/issues/3953

mister-roboto commented:
@cihanandac thanks for creating this Pull Request and helping to improve Plone!

TL;DR: Finish pushing changes, pass all other checks, then paste a comment:
```
@jenkins-plone-org please run jobs
```

To ensure that these changes do not break other parts of Plone, the Plone test suite matrix needs to pass, but it takes 30-60 min.  Other CI checks are usually much faster and the Plone Jenkins resources are limited, so when done pushing changes and all other checks pass either [start all Jenkins PR jobs yourself](https://jenkinsploneorg.readthedocs.io/en/latest/run-pull-request-jobs.html#run-a-pull-request-job), or simply add the comment above in this PR to start all the jobs automatically.

Happy hacking!

ale-rt commented:
@jenkins-plone-org please run jobs

ale-rt commented:
@jenkins-plone-org please run jobs

mauritsvanrees commented on plone/app/layout/viewlets/info.py:
```diff
@@ -0,0 +1,37 @@
+
+@implementer(IViewlet)
+class LockInfoViewlet(BrowserView):
+    """This is a viewlet which is not hooked up anywhere. It is referenced
+    from plone.app.layout. We do it this way to avoid having the  lower-level
+    plone.locking depend on these packages, whilst still providing
+    an implementation of the info box in a single place.
+    """
```
This comment is now outdated.  I think we can simply remove it.

mauritsvanrees reviewed (commented):
Can we rename `info.pt/py` to `locking.pt/py` please?

mauritsvanrees commented:
I have made maintenance branches for plone.app.layout and plone.locking and have updated coredev 6.1 to use them.

cihanandac commented:
- Renamed info.py/pt to locking.py/pt
- Removed the redundant comment
- Utilized the relative notion for the configure.zcml

Thanks for all the suggestions @mauritsvanrees, @petschki

petschki commented on news/3953.breaking:
```diff
@@ -0,0 +1 @@
+Moved plone.locking.browser.info.LockInfoViewlet template to plone.app.layout.viewlets @cihanandac
```
@cihanandac 🔝

cihanandac replied:
Thanks for catching this :)

mauritsvanrees commented:
Thanks!  I merged this one and your accompanying `plone.locking` PR.
